x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Reverse.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.string;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.compute.ann.Evaluator;
+import org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.expression.function.scalar.UnaryScalarFunction;
+
+import java.io.IOException;
+import java.text.BreakIterator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Function;
+
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.DEFAULT;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isString;
+
+public class Reverse extends UnaryScalarFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Reverse", Reverse::new);
+
+    private final Expression field;
+
+    @FunctionInfo(
+        returnType = { "keyword", "text" },
+        description = "Returns a new string representing the input string in reverse order.",
+        examples = {
+            @Example(file = "string", tag = "reverse"),
+            @Example(
+                file = "string",
+                tag = "reverseEmoji",
+                description = "`REVERSE` works with unicode, too! It keeps unicode grapheme clusters together during reversal."
+            ) }
+    )
+    public Reverse(
+        Source source,
+        @Param(
+            name = "str",
+            type = { "keyword", "text" },
+            description = "String expression. If `null`, the function returns `null`."
+        ) Expression field
+    ) {
+        super(source, field);
+        this.field = field;
+    }
+
+    private Reverse(StreamInput in) throws IOException {
+        this(Source.EMPTY, in.readNamedWriteable(Expression.class));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(field());
```

Comment:
Add `source.writeTo(out)` and read it with `this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class)`. Then you'll need to flip the test - it is setup with that `alwaysempty` thing.

After that the source should appear.

This is a left over from when serializing `Source` was really expensive. Back then we didn't serialize it. We should serialize it because it's really cheap now - just a few bytes.

Reply:
I realized `UnaryScalarFunction` already takes care of this, so I simplified the constructor in https://github.com/elastic/elasticsearch/pull/113297/commits/ad7c68fdb3b5c69e5b691652ddf587c965426adf